pkg/lint/rules/template.go

```diff
@@ -32,7 +32,7 @@ import (
 )
 
 // Templates lints the templates in the Linter.
-func Templates(linter *support.Linter, values []byte, namespace string, strict bool) {
+func Templates(linter *support.Linter, values []byte, namespace string) {
```

Comment:
Is this a backwards compatibility change? If so then the change should only be the removal of line as you did below:

```yaml
if strict {	
    e.Strict = true	
}
```

Maybe, just make changes similar to #8017?

Reply:
I agree. It is okay to leave a no-op param in the name of preserving API compatibility.

Reply:
oh, you're right, I wasn't thinking about any external usage. I agree that it does make more sense to keep the param. 

however, is the change really still "backwards compatible" if someone using the library function with the current logic, and is still expecting `e.Strict = true` to be set? I mean to say that such a change _should_ break the api. but the benefits for doing so in `v2.x` don't outweigh the negative impact of breaking the api, so i'll edit the PR to keep the param.